workflows/cloudify_system_workflows/deployment_environment.py

```diff
@@ -147,27 +148,34 @@ def delete(ctx, **kwargs):
     graph = ctx.graph_mode()
     sequence = graph.sequence()
 
-    sequence.add(
-        # uninstalling the operations worker
-        ctx.send_event('Stopping deployment operations worker'),
-        ctx.execute_task(
+    tasks = {
+        'stop_operations_worker': ctx.execute_task(
             task_name='cloudify_agent.installer.operations.stop'),
-        ctx.send_event('Deleting deployment operations worker'),
-        ctx.execute_task(
+        'delete_operations_worker': ctx.execute_task(
             task_name='cloudify_agent.installer.operations.delete'),
-
-        # uninstalling the workflows worker
-        ctx.send_event('Stopping deployment workflows worker'),
-        ctx.execute_task(
+        'stop_workflows_worker': ctx.execute_task(
             task_name='cloudify_agent.installer.operations.stop',
             kwargs=WORKFLOWS_WORKER_PAYLOAD),
-        ctx.send_event('Deleting deployment workflows worker'),
-        ctx.execute_task(
+        'delete_workflows_worker': ctx.execute_task(
             task_name='cloudify_agent.installer.operations.delete',
-            kwargs=WORKFLOWS_WORKER_PAYLOAD))
+            kwargs=WORKFLOWS_WORKER_PAYLOAD),
+        'stop_deployment_policy_engine_core':
+            ctx.execute_task('riemann_controller.tasks.delete')
+    }
+
+    # quick fix for unblocking botched deployments in 3.3.x
```

Comment:
i'd say the comment should have the word "delete"/"deletion" somewhere in there
it sounds like the deployment's stuck, but it's the deletion that fails